paasta_tools/paasta_remote_run_2.py

```diff
@@ -0,0 +1,101 @@
+import json
+from time import sleep
+
+from kubernetes.client.exceptions import ApiException
+
+from paasta_tools.kubernetes.application.controller_wrappers import (
+    get_application_wrapper,
+)
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.kubernetes_tools import load_kubernetes_service_config_no_cache
+from paasta_tools.utils import DEFAULT_SOA_DIR
+
+
+def remote_run_start(service, instance, user, cluster, interactive, recreate):
+    # TODO Overriding the kube client config for now as the api has limited permissions
+    kube_client = KubeClient(config_file="/etc/kubernetes/admin.conf")
+
+    # TODO hardcoded for now
+    is_eks = False
+
+    # Load the service deployment settings
+    deployment = load_kubernetes_service_config_no_cache(
+        service, instance, cluster, DEFAULT_SOA_DIR
+    )
+    namespace = deployment.get_namespace()
+
+    # Set to interactive mode
+    if interactive:
+        deployment.config_dict["cmd"] = "sleep 604800"  # One week
+
+    # Create the app with a new name
+    formatted_application = deployment.format_kubernetes_app()
+    formatted_application.metadata.name += f"-remote-run-{user}"
+    deployment_name = formatted_application.metadata.name
+    app_wrapper = get_application_wrapper(formatted_application)
+    app_wrapper.load_local_config(DEFAULT_SOA_DIR, cluster, is_eks)
+
+    # Launch pod
+    try:
+        app_wrapper.create(kube_client)
+    except ApiException as e:
+        if e.status == 409:
+            # Deployment already running
+            if recreate:
+                app_wrapper.deep_delete(kube_client)
+                wait_until_deployment_gone(kube_client, namespace, deployment_name)
+                app_wrapper.create(kube_client)
+            else:
+                raise
+
+    pod = wait_until_pod_running(kube_client, namespace, deployment_name)
+
+    return json.dumps(
+        {"status": "success", "pod_name": pod.metadata.name, "namespace": namespace}
+    )
+
+
+def wait_until_deployment_gone(kube_client, namespace, deployment_name):
+    for retry in range(10):
+        pod = find_pod(kube_client, namespace, deployment_name, 1)
+        if not pod:
+            return
+        sleep(5)
+    raise Exception("Pod still exists!")
```

Comment:
This doesn't work. The request times out before the deployment terminates. I think we'd need to return and the client needs to wait and retry.